src/coreComponents/physicsSolvers/contact/ContactSolverBase.cpp

```diff
@@ -116,6 +119,43 @@ void ContactSolverBase::setFractureRegions( dataRepository::Group const & meshBo
                  InputError );
 }
 
+real64 ContactSolverBase::solverStep( real64 const & time_n,
+                                      real64 const & dt,
+                                      const integer cycleNumber,
+                                      DomainPartition & domain )
+{
+  if( cycleNumber == 0 )
```

Comment:
![image](https://github.com/user-attachments/assets/370c6661-cf10-4f12-b379-209d5c68fe12)

Reply:
we really need to work on that additional initialization stage.

Reply:
at least with this implementation it really only applies the `initialConditions` to the fractures.

Reply:
@CusiniM would that work when you have mechanics initialization step running as `cycleNumber == 0`?

Reply:
Certainly not but we need to first figure out how to initialize problems with faults and then abstract the way initialization is performed. 

In any case this solver is only used for Rate and state atm so I am not too worried about it. We could move it to SurfaceGenerator::execute though and then I don't think it would ever be an issue.